eis_toolkit/prediction/weights_of_evidence.py

```diff
@@ -0,0 +1,255 @@
+
+    class_2_w_gen = np.log(class_2_point_count / total_deposits) - np.log(
+        (class_2_count - class_2_point_count) / total_no_deposits
+    )
+    clas_2_s_wpls_gen = np.sqrt((1 / class_2_point_count) + (1 / (class_2_count - class_2_point_count)))
+
+    df["Generalized W+"] = round(class_2_w_gen, 4)
+    df["Generalized S_W+"] = round(clas_2_s_wpls_gen, 4)
+
+    # Class 1
+    class_1_count = df.loc[len(df.index) - 1, "Pixel count"] - class_2_count
+    class_1_point_count = df.loc[len(df.index) - 1, "Deposit count"] - class_2_point_count
+
+    class_1_w_gen = np.log(class_1_point_count / total_deposits) - np.log(
+        (class_1_count - class_1_point_count) / total_no_deposits
+    )
+    clas_1_s_wpls_gen = np.sqrt((1 / class_1_point_count) + (1 / (class_1_count - class_1_point_count)))
+    df.loc[df["Generalized class"] == 1, "Generalized W+"] = round(class_1_w_gen, 4)
+    df.loc[df["Generalized class"] == 1, "Generalized S_W+"] = round(clas_1_s_wpls_gen, 4)
+
+
+def _generate_rasters_from_metrics(
+    evidence: np.ndarray, df: pd.DataFrame, metrics_to_include: List[str] = ["Class", "W+", "S_W+"]
+) -> dict:
+    """Generate rasters for defined metrics based."""
+    raster_dict = {}
+    for metric in metrics_to_include:
+        raster = np.full(evidence.shape, np.nan)
+        for _, row in df.iterrows():
+            mask = np.isin(evidence, row["Class"])
+            raster[mask] = row[metric]
+        raster_dict[metric] = raster
+    return raster_dict
+
+
+@beartype
+def weights_of_evidence(
+    evidential_raster: rasterio.io.DatasetReader,
+    deposits: gpd.GeoDataFrame,
+    raster_nodata: Optional[Number] = None,
+    weights_type: Literal["unique", "ascending", "descending"] = "unique",
+    studentized_contrast_threshold: Number = 2,
+    rasters_to_generate: Sequence[str] = ["Class", "W+", "S_W+"],
+) -> Tuple[pd.DataFrame, dict, dict]:
+    """
+    Calculate weights of spatial associations.
+
+    Args:
+        evidential_raster: The evidential raster.
+        deposits: Vector data representing the mineral deposits or occurences point data.
+        raster_nodata: If nodata value of raster is wanted to specify manually. Optional parameter, defaults to None
+            (nodata from raster metadata is used).
+        weights_type: Accepted values are 'unique' for unique weights, 'ascending' for cumulative ascending weights,
+            'descending' for cumulative descending weights. Defaults to 'unique'.
+        studentized_contrast_threshold: Studentized contrast threshold value used to reclassify all classes.
+            Reclassification is used when creating generalized rasters with cumulative weight type selection.
+            Not needed if weights_type is 'unique'. Defaults to 2.
+        rasters_to_generate: Rasters to generate from the computed weight metrics. All column names
+            in the produced weights_df are valid choices. Defaults to ["Class", "W+", "S_W+]
```

Comment:
```suggestion
            in the produced weights_df are valid choices. Defaults to weights_of_evidence.DEFAULT_METRICS
```

If the earlier suggestion is implemented, this way you could reduce repeated explicit string typing even in documentation. Mistakes happen easily when strings are explicitly repeated. Even though this will not show in the literal documentation the values, usually a Python user can then just print our the values by himself.

Reply:
I used the static variables for column names in the code, but I would prefer to keep the documentation as clear as possible and therefore wanted to display the actual column names in the docstring.

Reply:
Yeah, this is a subjective question. Sounds good!